src/mmw/js/src/app.js

```diff
@@ -64,6 +65,34 @@ var App = new Marionette.Application({
 
     hideLoginModal: function() {
         $('#login').modal('hide');
+    },
+
+    userLogout: function() {
+        var csrftoken = helpers.Cookie.get('csrftoken');
+
+        $.ajax({
+            url: '/user/ajaxlogout',
+            type: 'GET',
+            beforeSend: function(xhr) {
+                xhr.setRequestHeader('X-CSRFToken', csrftoken);
+            }
+        }).always(function() {
+            // We silence the first change because we don't want to send two
+            // events. The event generated by the second change is enough.
+            App.user.unset('username', {silent: true});
+            App.user.set('guest', true);
+
+            // Let the user stay at `/` or `/analyze`, but other pages require
+            // an account so in those cases redirect to `/`
+            var router = require('./router').router;
+            var currentRoute = Backbone.history.getFragment();
+            if (currentRoute === '' || currentRoute === 'analyze') {
```

Comment:
Do we have a mechanism for identifying views that are logged-in user only? This if-condition seems to be a really flimsy place to list them.